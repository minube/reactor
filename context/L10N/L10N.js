import { node, shape, string } from 'prop-types';
import React, { createContext, PureComponent } from 'react';

const { Provider, Consumer: ConsumerL10N } = createContext('reactor:l10n');

class ProviderL10N extends PureComponent {
  static propTypes = {
    children: node,
    dictionary: shape({}),
    language: string,
  };

  static defaultProps = {
    children: undefined,
    dictionary: {},
    language: 'en-EN',
  };

  render() {
    const { props: { children, dictionary, language } } = this;

    return (
      <Provider value={{ l10n: dictionary[language] || {} }}>
        { children }
      </Provider>
    );
  }
}

export { ConsumerL10N, ProviderL10N };
