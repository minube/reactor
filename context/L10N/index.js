export { ConsumerL10N, ProviderL10N } from './L10N';
