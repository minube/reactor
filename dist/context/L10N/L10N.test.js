import React from 'react';
import TestRenderer from 'react-test-renderer';

import { ConsumerL10N, ProviderL10N } from './L10N';

global.fetch = () => {};

describe('<ProviderL10N>', () => {
  it('renders', () => {
    const tree = TestRenderer.create(<ProviderL10N />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

