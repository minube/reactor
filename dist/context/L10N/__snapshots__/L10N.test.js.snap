// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProviderL10N> renders 1`] = `null`;
